### DISCUSSION TITLE
How can I disable streaming output for tools/steps with a provider not supporting streaming?

### DISCUSSION THREAD

**Discussion by crazybullet28** [langgenius/dify#7109] [Help]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

I'm trying to add a new model provider, but that provider currently didn't support streaming output. If I directly return the LLMResult type for the `_invoke` function, it will return error `Run failed: Node LLM run failed: 'tuple' object has no attribute 'delta'` when calling the `_handle_invoke_result function` in api/core/workflow/nodes/llm/llm_node.py, as the returned LLMResult object didn't include the 'delta' fields.

Here is the returned LLMResult object:
`LLMResult(model='gpt-4', prompt_messages=[UserPromptMessage(role=<PromptMessageRole.USER: 'user'>, content='hi', name=None)], message=AssistantPromptMessage(role=<PromptMessageRole.ASSISTANT: 'assistant'>, content='Hello! How can I help you today?', name=None, tool_calls=[]), usage=LLMUsage(prompt_tokens=8, prompt_unit_price=Decimal('0.0'), prompt_price_unit=Decimal('0.0'), prompt_price=Decimal('0.0'), completion_tokens=9, completion_unit_price=Decimal('0.0'), completion_price_unit=Decimal('0.0'), completion_price=Decimal('0.0'), total_tokens=17, total_price=Decimal('0.0'), currency='USD', latency=21.528332874993794), system_fingerprint=None)`

Is there any way I can disable the streaming output for certain workflows / tools / steps? Or how should I modify my provider model code?

**Comment by lzj1991** [comment]
Has it been solved? I'm encountering the same issue.

**Reply by crazybullet28** [reply]
Hi, I have not found a correct solution yet. Our temporary workaround is to add a wrapper to the model provider, making it appear as though it returns a streaming output (even though the complete result has already been returned). This approach is certainly not elegant and introduces significant latency. Also happy to hear if you have any other solutions.

**Reply by warm200** [reply]
do you mind sharing how you implement this?
